code/game/machinery/doors/airlock.dm

```diff
@@ -314,6 +314,26 @@
 		note = null
 		update_icon()
 
+/obj/machinery/door/airlock/Bumped(atom/movable/AM)
+	if(operating || (obj_flags & EMAGGED))
+		return
+	if(ismecha(AM))
+		var/obj/mecha/mecha = AM
+		if(density)
```

Comment:
Is it even possible to bump an object with no density (Nothing needs changing, just wondering)